src/views/Releases/Release/index.jsx

```diff
@@ -1,11 +1,163 @@
-import React from 'react';
-import Typography from '@material-ui/core/Typography';
+import React, { Fragment, useState, useEffect } from 'react';
+import classNames from 'classnames';
+import Spinner from '@mozilla-frontend-infra/components/Spinner';
+import { makeStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import SpeedDialAction from '@material-ui/lab/SpeedDialAction';
+import Tooltip from '@material-ui/core/Tooltip';
+import Fab from '@material-ui/core/Fab';
+import ContentSaveIcon from 'mdi-react/ContentSaveIcon';
+import DeleteIcon from 'mdi-react/DeleteIcon';
 import Dashboard from '../../../components/Dashboard';
+import ErrorPanel from '../../../components/ErrorPanel';
+import SpeedDial from '../../../components/SpeedDial';
+import AutoCompleteText from '../../../components/AutoCompleteText';
+import CodeEditor from '../../../components/CodeEditor';
+import useAction from '../../../hooks/useAction';
+import { getReleases, getRelease } from '../../../services/releases';
+import { getProducts } from '../../../services/rules';
+import getSuggestions from '../../../components/AutoCompleteText/getSuggestions';
+
+const useStyles = makeStyles(theme => ({
+  fab: {
+    ...theme.mixins.fab,
+  },
+  secondFab: {
+    ...theme.mixins.fab,
+    right: theme.spacing(12),
+  },
+  // Ensure fab  has higher priority than the code editor.
+  // Otherwise, the fab is not clickable.
+  saveButton: {
+    zIndex: 10,
+  },
+}));
+
+export default function Release(props) {
+  const classes = useStyles();
+  const {
+    isNewRelease,
+    match: {
+      params: { releaseName },
+    },
+  } = props;
+  const [releaseNameValue, setReleaseNameValue] = useState(releaseName || '');
+  const [productTextValue, setProductTextValue] = useState('');
+  const [releaseEditorValue, setReleaseEditorValue] = useState('{}');
+  const [release, fetchRelease] = useAction(getRelease);
+  const fetchReleases = useAction(getReleases)[1];
+  const [products, fetchProducts] = useAction(getProducts);
+  const isLoading = release.loading || products.loading;
+  // TODO: Fill actionLoading when hooking up mutations
+  const actionLoading = false;
+  const error = release.error || products.error;
+
+  useEffect(() => {
+    if (releaseName) {
+      Promise.all([fetchRelease(releaseName), fetchReleases()]).then(
+        ([fetchedRelease, fetchedReleases]) => {
+          if (fetchedRelease.data) {
+            setReleaseEditorValue(JSON.stringify(fetchedRelease.data, null, 2));
+          }
+
+          if (fetchedReleases.data) {
+            const r = fetchedReleases.data.data.releases.find(
+              r => r.name === releaseName
+            );
+
+            setProductTextValue(r.product);
+          }
+        }
+      );
+    }
+
+    fetchProducts();
+  }, [releaseName]);
```

Comment:
Given that `releaseName` is static (as far as I can tell, at least), does this need to depend on it? I'm guessing this serves a purpose that I'm not aware of, but because it makes XHR requests that don't need to be done more than once, I'm curious.

Reply:
It would probably still work in this case but we are strongly recommended to mention all of the dependencies that the effect depends on. This would avoid any weird errors like seeing a previous value instead of the latest one. From https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect:

> every value referenced inside the effect function should also appear in the dependencies array. In the future, a sufficiently advanced compiler could create this array automatically.

Reply:
OK, makes sense. And since the dependency is effectively constant, this shouldn't ever refire?

Reply:
Correct.